internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,251 @@
+		}
+
+		if r.YearlyBackup != nil && *r.YearlyBackup > 0 {
+			costComponents = append(costComponents, r.sqlMIYearlyBackupCostComponent())
+		}
+	}
+	return costComponents
+}
+
+func (r *SQLManagedInstance) productDescription() *string {
+
+	productDescription := ""
+	if strings.Contains(r.SKU, "GP") {
+		productDescription = "SQL Managed Instance General Purpose"
+	} else if strings.Contains(r.SKU, "BC") {
+		productDescription = "SQL Managed Instance Business Critical"
+	}
+
+	if strings.Contains(r.SKU, "Gen5") {
+		productDescription = fmt.Sprintf("%s - %s", productDescription, "Compute Gen5")
+	}
+	return strPtr(productDescription)
+}
+
+func (r *SQLManagedInstance) meteredName() *string {
+	meterName := fmt.Sprintf("%s %s", strconv.FormatInt(*r.Cores, 10), "vCore")
+	return strPtr(meterName)
+}
+
+func (r *SQLManagedInstance) storageCost() *schema.CostComponent {
+
+	storageComponent := schema.CostComponent{}
+	storageComponent.Name = fmt.Sprintf("Storage %s Gb (first 32 Gb include)", strings.ToTitle(strconv.FormatInt(*r.StorageSizeInGb, 10)))
+	storageComponent.Unit = "Unit of 32Gb"
+	storageComponent.UnitMultiplier = decimal.NewFromInt(1)
+	storageComponent.MonthlyQuantity = decimalPtr(decimal.NewFromInt(*r.StorageSizeInGb - 32))
+	storageComponent.ProductFilter = &schema.ProductFilter{
+		VendorName:    strPtr(vendorName),
+		Region:        strPtr(r.Region),
+		Service:       strPtr(sqlMIServiceName),
+		ProductFamily: strPtr(sqlMIProductFamily),
+		AttributeFilters: ([]*schema.AttributeFilter{
+			{Key: "productName", Value: strPtr("SQL Managed Instance General Purpose - Storage")},
+			{Key: "meterName", Value: strPtr("Data Stored")},
+		}),
+	}
+
+	storageComponent.PriceFilter = priceFilterConsumption
+	return &storageComponent
+}
+
+func (r *SQLManagedInstance) backupCost() *schema.CostComponent {
+	backupCostComponent := schema.CostComponent{}
+	backupCostComponent.Name = fmt.Sprintf("Backup Cost for %s Gb %s", strconv.FormatInt(*r.BackupStorageGb, 10), r.StorageAccountType)
+	backupCostComponent.Unit = "Gb"
+	backupCostComponent.UnitMultiplier = decimal.NewFromInt(1)
+	backupCostComponent.MonthlyQuantity = decimalPtr(decimal.NewFromInt(*r.BackupStorageGb))
+	backupCostComponent.ProductFilter = &schema.ProductFilter{
+		VendorName:    strPtr(vendorName),
+		Region:        strPtr(r.Region),
+		Service:       strPtr(sqlMIServiceName),
+		ProductFamily: strPtr(sqlMIProductFamily),
+		AttributeFilters: ([]*schema.AttributeFilter{
+			{Key: "productName", Value: strPtr("SQL Managed Instance PITR Backup Storage")},
+			{Key: "meterName", Value: strPtr(fmt.Sprintf("%s Data Stored", r.StorageAccountType))},
+		}),
+	}
+	backupCostComponent.PriceFilter = priceFilterConsumption
+	return &backupCostComponent
+}
+
+func (r *SQLManagedInstance) sqlMILicenseCostComponent() *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:           "SQL license",
+		Unit:           "vCore-hours",
```

Comment:
This needs to be changed to something like `licence` and the `UnitMultiplier` should have `schema.HourToMonthUnitMultiplier` to solve outputting `730` in the cost output.

Reply:
done